bool tryingToLauch
bool launched
bool inRange
bool fail
string failMsg